File: sha256tokengenerator.php
<?php
function sign_token($url_prefix, $base64_key, $algo, $expiration_time = null) {
    // Gets the Signed URL Suffix string for the Media CDN' Short token URL requests.

    $output = 'URLPrefix=' . base64_encode($url_prefix);

    if (!$expiration_time) {
        $expiration_time = new DateTime();
        $expiration_time->add(new DateInterval('PT1H')); // Default to 1 hour if not provided
    }
    $epoch_duration = $expiration_time->getTimestamp();
    $output .= '~Expires=' . $epoch_duration;

    $key = base64_decode($base64_key);
    $algo = strtolower($algo);
    if ($algo == 'sha1') {
        $signature = hash_hmac('sha1', $output, $key);
    } elseif ($algo == 'sha256') {
        $signature = hash_hmac('sha256', $output, $key);
    } else {
        throw new Exception('User input(`algo`) can be either `sha1` or `sha256`');
    }
    $output .= '~hmac=' . $signature;
    return $output;
}

// Ask user for inputs
echo "Enter the URL prefix (e.g., https://vod.cdntest.in/tearsofsteel/): ";
$up = trim(fgets(STDIN));

echo "Enter the base64 key (e.g., LGRWQgV4nlRIgjggklkhjkhljlkjkl;hkg91MuY=): ";
$bk = trim(fgets(STDIN));

echo "Enter expiration time in seconds (default is 60): ";
$expiration_seconds_input = trim(fgets(STDIN));
$expiration_seconds = $expiration_seconds_input ? intval($expiration_seconds_input) : 60;

echo "Enter the complete URL (e.g., https://vod.cdntest.in/tearsofsteel/manifest.m3u8): ";
$url_input = trim(fgets(STDIN));

// Calculate expiration time
$expiration_time = new DateTime();
$expiration_time->add(new DateInterval('PT' . $expiration_seconds . 'S'));

// Generate the signed token
$x = sign_token($up, $bk, 'sha256', $expiration_time);
$xb = strval($x);
$y = $url_input . '?hdnts=' . $xb;
echo $y . PHP_EOL;

// Make a request to the generated URL and print the response code and headers
$response = file_get_contents($y, false, stream_context_create([
    'http' => [
        'method' => 'GET',
    ]
]));

$http_response_header = isset($http_response_header) ? $http_response_header : [];

echo $response . PHP_EOL;
echo "Response Code: " . parse_response_code($http_response_header) . PHP_EOL;
echo "Response Headers:" . PHP_EOL;
foreach ($http_response_header as $header) {
    echo $header . PHP_EOL;
}

function parse_response_code($headers) {
    foreach ($headers as $header) {
        if (preg_match('/^HTTP\/\d+\.\d+\s+(\d+)/', $header, $matches)) {
            return intval($matches[1]);
        }
    }
    return 0;
}
?>

File: sha256tokengenerator.py
#Python3 script to generate sha256 tokens
import base64
import datetime
import hashlib
import hmac
import requests

def sign_token(url_prefix: str, base64_key: bytes, algo: str, expiration_time: datetime.datetime = None) -> bytes:
    """Gets the Signed URL Suffix string for the Media CDN' Short token URL requests.
    Args:
        url_prefix: URL prefix to sign as a string.
        base64_key: Secret key as a base64 encoded string.
        algo: Algorithm can be either `SHA1` or `SHA256`.
        expiration_time: Expiration time as a UTC datetime object.
    Returns:
        Returns the Signed URL appended with the query parameters based on the
        specified URL prefix and configuration.
    """
    output = b"URLPrefix=" + base64.standard_b64encode(url_prefix.encode("utf-8"))

    if not expiration_time:
        expiration_time = datetime.datetime.now() + datetime.timedelta(hours=1)
    epoch_duration = int(
        (expiration_time - datetime.datetime.utcfromtimestamp(0)).total_seconds()
    )
    output += b"~Expires=" + str(epoch_duration).encode("utf-8")
    key = base64.standard_b64decode(base64_key)
    algo = algo.lower()
    if algo == "sha1":
        signature = hmac.new(key, output, digestmod=hashlib.sha1).hexdigest()
    elif algo == "sha256":
        signature = hmac.new(key, output, digestmod=hashlib.sha256).hexdigest()
    else:
        raise ValueError("User input(`algo`) can be either `sha1` or `sha256`")
    output += b"~hmac=" + signature.encode("utf-8")
    return output

# Ask user for inputs
up = input("Enter the URL prefix (e.g., https://vod.cdntest.in/tearsofsteel/): ")
bk = input("Enter the base64 key (e.g., LGRWJKHKHKK76S7S7H7R8KHJHKJHJHGG1MuY=): ")

expiration_seconds_input = input("Enter expiration time in seconds (default is 60): ")
expiration_seconds = int(expiration_seconds_input) if expiration_seconds_input else 60

url_input = input("Enter the complete URL (e.g., https://vod.cdntest.in/tearsofsteel/manifest.m3u8): ")

date_time_str = str(datetime.datetime.now() + datetime.timedelta(seconds=expiration_seconds))
date_time_obj = datetime.datetime.strptime(date_time_str, '%Y-%m-%d %H:%M:%S.%f')
x = sign_token(up, bk.encode(), "SHA256", date_time_obj)
xb = str(x)
y = f"{url_input}?hdnts={xb[2:-1]}"
print(y)

# Make a request to the generated URL and print the response code and headers
response = requests.get(y)

# Print the response content
print(response.content)

# Print the response code
print("Response Code:", response.status_code)

# Print the response headers
print("Response Headers:")
for header, value in response.headers.items():
    print(f"{header}: {value}")
